test/actions.test.ts

```diff
@@ -77,84 +66,113 @@ describe("Test Custom SDK Logic", () => {
-        stakingRequests: "disabled",
-        isBiddable: false,
-        gridViewByDefault: false,
-        customDomainHeader: false,
-        previewImage: "preview_image",
-        customDomainHeaderValue: "custom_domain",
-      };
-      const registrar: Registrar = await getRegistrar(
-        provider,
-        registrarAddress
-      );
-      const apiUri = "https://zns.api.zero.tech/api";
-      const client = createClient(apiUri);
-      const tx = await actions.setDomainMetadata(
-        kovanDomainId,
-        metadata,
-        client,
-        signer,
-        registrar
-      );
-      console.log(tx);
-      const retrievedMetadata = await actions.getDomainMetadata(
-        kovanDomainId,
-        registrar,
-        IPFSGatewayUri.fleek
-      );
-      expect(metadata).deep.equal(retrievedMetadata);
-    });
+    // Keep as an example call, but comment it
+    // it("runs setDomainMetadata", async () => {
+    //   const registrar: Registrar = await getRegistrar(
+    //     provider,
+    //     registrarAddress
+    //   );
+    //   const lockedStatus = await registrar.isDomainMetadataLocked(domainId);
+    //   // If locked, call to unlock before calling to set
+    //   if(lockedStatus) {
+    //     const tx = await registrar.connect(signer).lockDomainMetadata(domainId, false);
+    //     await tx.wait(5); // Wait for unlock confirmation before trying to set metadata
+    //   }
+    //   const metadata = await actions.getDomainMetadata(
+    //     domainId,
+    //     registrar,
+    //     IPFSGatewayUri.fleek);
+    //   // Set to a new value every time it's run,
+    //   // still 1/100 chance it's the same metadata and that will fail
+    //   const rand = Math.round(Math.random() * 100);
+    //   metadata.description = `A random number is: ${rand}`
+    //   const client = createClient(config.apiUri);
+    //   const tx = await actions.setDomainMetadata(
+    //     domainId,
+    //     metadata,
+    //     client,
+    //     signer,
+    //     registrar
+    //   );
+    //   console.log(tx.v);
+    //   const retrievedMetadata = await actions.getDomainMetadata(
+    //     domainId,
+    //     registrar,
+    //     IPFSGatewayUri.fleek
+    //   );
+    //   expect(metadata).deep.equal(retrievedMetadata);
+    // });
   });
-  describe("getBuyNowPrice", () => {
+  describe("(get|set)buyNowPrice", () => {
     it("runs as expected", async () => {
       const zAuctionRouteUriToInstance = createZAuctionInstances(config);
 
       const zAuctionInstance = await getZAuctionInstanceForDomain(
-        kovanDomainId,
+        domainId,
         config.zAuctionRoutes,
         zAuctionRouteUriToInstance,
         domainIdToDomainName
       );
-      const params: BuyNowParams = {
-        amount: ethers.utils.parseEther("1.0").toString(),
-        tokenId: kovanDomainId,
+      // Set to a new value every time it's run,
+      // still 1/100 chance it's the same price and that will fail
```

Comment:
could we just exclude the current price from the possible random prices? set it to 100 and do `Math.round(Math.random() * 99);` maybe?

Reply:
The issue is still the same in any situation where you run it more than once. If we set to 100, we can range 0-99, but then the next time we run it the value is something between 0-99, say 50, now we can only use values 0-49 or 51-100, and so on. I suppose a way around this would be to get the price and generate numbers until the new price is not that price, then it will never conflict